File: php/src/Model/AssignTaskResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The result of the operator override `tasks()->assign()`.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class AssignTaskResult
{
    public function __construct(
        public readonly string $id,
        public readonly string $status,
        public readonly string $workerId,
        /** null when the task came from the queue */
        public readonly ?string $previousWorkerId,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            workerId: (string) ($data['workerId'] ?? ''),
            previousWorkerId: isset($data['previousWorkerId']) ? (string) $data['previousWorkerId'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/AssignTaskResult.java
package io.fivexer.sdk.model;

/**
 * The result of the operator override {@code tasks().assign(...)}.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class AssignTaskResult {
    private String id;
    private String status;
    private String workerId;
    private String previousWorkerId;  // the worker who held the task before a reassignment; null when it came from the queue

    public String getId() { return id; }
    public String getStatus() { return status; }
    public String getWorkerId() { return workerId; }
    public String getPreviousWorkerId() { return previousWorkerId; }
}
